### PR TITLE
use node attrs for data_bag name/item

### PR DESCRIPTION
**Pull request by curiositycasualty**

### CODE REVIEW

**Comment by manuasir** [comment]
Hello @curiositycasualty ,

Can you please provide more details about this PR?
A description of the new functionality/fix and how to use would be great.

Thanks

File: wazuh/recipes/wazuh_api.rb [hunk 1]
```diff
@@ -26,7 +26,10 @@
 package ['nodejs', 'wazuh-api']
 
 
-api_keys = Chef::EncryptedDataBagItem.load('wazuh_secrets', 'api')
+api_keys = Chef::EncryptedDataBagItem.load(
+  node['wazuh']['api']['data_bag']['name'],
+  node['wazuh']['api']['data_bag']['item']
+)
```
Comment (devsibwarra) [inline wazuh/recipes/wazuh_api.rb]: 👍 for the ability to override data bag settings

Can you replace `Chef::EncryptedDataBagItem.load(...)` with `data_bag_item(...)`?  Docs for that are [here](https://docs.chef.io/data_bags.html)

> If `item` is encrypted, `data_bag_item` will automatically decrypt it using the key specified above, or (if none is specified) by the `Chef::Config[:encrypted_data_bag_secret]` method, which defaults to `/etc/chef/encrypted_data_bag_secret`.

**Comment by curiositycasualty** [comment]
@manuasir This turns the strings used to identify the data_bag name and item into node attributes, subject to attribute inheritance and therefore alterable in Chef environments, roles, wrapper cookbooks, etc. When previously they were statically set to just "wazuh_secerts" and "api".

Those are still the defaults as set through `attributes/default.rb`.

**Comment by curiositycasualty** [comment]
Oh, it looks like there's `Chef::EncryptedDataBagItem.load`s all over the place :(

**Pull request closed without merging**